File: scripts/parse_routes.js
const fs = require('fs');
const path = require('path');

const routesDir = path.join(__dirname, '../src/app/module');
const routeMap = [];

const routePattern = /\.(get|post|patch|put|delete)\(\s*(["'][^"']+["'])/g;

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (file.endsWith('.routes.ts')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            const moduleName = file.replace('.routes.ts', '');
            let match;
            while ((match = routePattern.exec(content)) !== null) {
                const method = match[1].toUpperCase();
                const routePath = match[2].replace(/['"]/g, '');
                const fullRoutePath = `/${moduleName}${routePath === '/' ? '' : routePath}`;
                routeMap.push({
                    method,
                    path: fullRoutePath,
                    file: fullPath
                });
            }
        }
    }
}

walkDir(routesDir);

fs.writeFileSync(path.join(__dirname, '../route_map.json'), JSON.stringify(routeMap, null, 2), 'utf8');
console.log(`Generated route_map.json with ${routeMap.length} endpoints.`);

File: scripts/parse_routes.py
import os
import re
import json

routes_dir = 'src/app/module'
route_map = []

# regex to find route definitions like: .get("/path", middleware1, middleware2, controller.method)
route_pattern = re.compile(r'\.(get|post|patch|put|delete)\(\s*(["\'][^"\']+["\'])')

for root, _, files in os.walk(routes_dir):
    for file in files:
        if file.endswith('.routes.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
                matches = route_pattern.findall(content)
                module_name = file.replace('.routes.ts', '')
                for method, route_path in matches:
                    route_path = route_path.strip('\'"')
                    full_path = f"/{module_name}{route_path}"
                    route_map.append({
                        "method": method.upper(),
                        "path": full_path,
                        "file": path
                    })

with open('route_map.json', 'w', encoding='utf-8') as f:
    json.dump(route_map, f, indent=2)

print(f"Generated route_map.json with {len(route_map)} endpoints.")
